src/js/test/rpc/server.test.ts

```diff
@@ -366,5 +370,164 @@ describe("Server", function () {
+      server.loadCoprocFromString = (_, _2) => createHandle().coprocessor;
+      server.listen(8080);
+      return SupervisorClient.create(8080).then((client) => {
+        client
+          .enable_coprocessors({
+            coprocessors: [{ id: BigInt(1), source_code: Buffer.from("") }],
+          })
+          .then((response) => {
+            response.responses.forEach((responseItem) => {
+              assert.strictEqual(
+                responseItem.enableResponseCode,
+                EnableResponseCodes.success
+              );
+              assert.strictEqual(
+                responseItem.scriptMetadata.id,
+                createHandle().coprocessor.globalId
+              );
+              const metaDataExpected: EnableCoprocessorMetadata = {
+                id: createHandle().coprocessor.globalId,
+                inputTopic: createHandle().coprocessor.inputTopics.map(
+                  (topic) => ({
+                    topic,
+                    ingestion_policy: 1,
+                  })
+                ),
+              };
+              assert.deepStrictEqual(
+                responseItem.scriptMetadata,
+                metaDataExpected
+              );
+            });
+          })
+          .finally(() => {
+            client.close();
+            server.closeConnection();
+          });
+      });
+    });
+
+    it(
+      "shouldn't enable coprocessor if the coprocessor doesn't " +
+        "have topics",
+      () => {
+        const server = new ProcessBatchServer();
+        const coprocessor = createHandle().coprocessor;
+        coprocessor.inputTopics = [];
+        server.loadCoprocFromString = (_, _2) => coprocessor;
+        server.listen(8080);
+        return SupervisorClient.create(8080).then((client) => {
+          client
+            .enable_coprocessors({
+              coprocessors: [{ id: BigInt(1), source_code: Buffer.from("") }],
+            })
+            .then((response) => {
+              response.responses.forEach((responseItem) => {
+                assert.strictEqual(
+                  responseItem.enableResponseCode,
+                  EnableResponseCodes.scriptContainsNoTopics
+                );
+                assert.strictEqual(
+                  responseItem.scriptMetadata.id,
+                  createHandle().coprocessor.globalId
+                );
+                const metaDataExpected: EnableCoprocessorMetadata = {
+                  id: createHandle().coprocessor.globalId,
+                  inputTopic: [],
+                };
+                assert.deepStrictEqual(
+                  responseItem.scriptMetadata,
+                  metaDataExpected
+                );
+              });
+            })
+            .finally(() => {
+              client.close();
+              server.closeConnection();
+            });
+        });
+      }
+    );
+
+    it("shouldn't enable coprocessor if the coprocessor has invalid topics", () => {
```

Comment:
If you just want to test the kafka validation you can just invoke the function, theres no need for all of the networking code

Reply:
Yes, but I like to have all flow because I test the client response, but I'll have it in mind for the test.

Reply:
I have unit tests and then I have full end to end tests. You still need your unit tests because they are more targeted, if there's an issue its immediately obvious.

Reply:
Done. created